internal/server/unlink_person.go

```diff
@@ -0,0 +1,71 @@
+package server
+
+import (
+	"github.com/ministryofjustice/opg-go-common/template"
+	"github.com/ministryofjustice/opg-sirius-lpa-frontend/internal/sirius"
+	"net/http"
+	"strconv"
+)
+
+type UnlinkPersonClient interface {
+	Person(sirius.Context, int) (sirius.Person, error)
+	UnlinkPerson(sirius.Context, int, int) error
+}
+
+type unlinkPersonData struct {
+	XSRFToken string
+	Success   bool
+	Error     sirius.ValidationError
+
+	Person sirius.Person
+}
+
+func UnlinkPerson(client UnlinkPersonClient, tmpl template.Template) Handler {
+	return func(w http.ResponseWriter, r *http.Request) error {
+		parentID, err := strconv.Atoi(r.FormValue("id"))
+		if err != nil {
+			return err
+		}
+
+		ctx := getContext(r)
+		data := unlinkPersonData{XSRFToken: ctx.XSRFToken}
+
+		data.Person, err = client.Person(ctx, parentID)
+		if err != nil {
+			return err
+		}
```

Comment:
you should do this after handling the post, that way when the form comes back you'll have the correct state of the children shown